File: app/src/main/java/com/example/myapplication/adapters/animeadapters/recycleradapters/RecyclerAllGenreAdapter.java
package com.example.myapplication.adapters.animeadapters.recycleradapters;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapplication.R;
import com.example.myapplication.databinding.ItemListSelectChapterBinding;
import com.example.myapplication.fragments.anime_fragments.genre_and_search_mvp.GenreAndSearchAnimeFragment;
import com.example.myapplication.models.animemodels.AnimeGenreAndSearchResultModel;

import java.util.List;

public class RecyclerAllGenreAdapter extends RecyclerView.Adapter<RecyclerAllGenreAdapter.ViewHolder> {
    private GenreAndSearchAnimeFragment context;
    private List<AnimeGenreAndSearchResultModel.AnimeGenreResult> allGenreList;
    private ClickGenreListener clickListener;

    public RecyclerAllGenreAdapter(GenreAndSearchAnimeFragment context, List<AnimeGenreAndSearchResultModel.AnimeGenreResult> allGenreList) {
        this.context = context;
        this.allGenreList = allGenreList;
        if (context != null) {
            clickListener = context;
        }
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(context.getContext());
        ItemListSelectChapterBinding itemListBinding = DataBindingUtil.inflate(layoutInflater, R.layout.item_list_select_chapter, parent, false);
        return new ViewHolder(itemListBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.itemListBinding.textViewChapterAll.setText(allGenreList.get(position).getGenreTitle());
        Log.e("positionFromAdapter", "" + position);
        holder.itemListBinding.linearViewChapterAll.setOnClickListener(v -> clickListener.onItemClickGenre(position));
    }

    @Override
    public int getItemCount() {
        return allGenreList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemListSelectChapterBinding itemListBinding;

        public ViewHolder(final ItemListSelectChapterBinding itemView) {
            super(itemView.getRoot());
            this.itemListBinding = itemView;
        }
    }

    public interface ClickGenreListener {
        void onItemClickGenre(int position);
    }
}

File: app/src/main/java/com/example/myapplication/adapters/animeadapters/recycleradapters/RecyclerAllGenreAdapter.kt
package com.example.myapplication.adapters.animeadapters.recycleradapters

import android.util.Log
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView

import com.example.myapplication.R
import com.example.myapplication.databinding.ItemListSelectChapterBinding
import com.example.myapplication.fragments.anime_fragments.genre_and_search_mvp.GenreAndSearchAnimeFragment
import com.example.myapplication.models.animemodels.AnimeGenreAndSearchResultModel

class RecyclerAllGenreAdapter(private val context: GenreAndSearchAnimeFragment?, private val allGenreList: List<AnimeGenreAndSearchResultModel.AnimeGenreResult>) : RecyclerView.Adapter<RecyclerAllGenreAdapter.ViewHolder>() {
    private var clickListener: ClickGenreListener? = null

    init {
        if (context != null) {
            clickListener = context
        }
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(context!!.context)
        val itemListBinding = DataBindingUtil.inflate<ItemListSelectChapterBinding>(layoutInflater, R.layout.item_list_select_chapter, parent, false)
        return ViewHolder(itemListBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.itemListBinding.textViewChapterAll.text = allGenreList[position].genreTitle
        Log.e("positionFromAdapter", "" + position)
        holder.itemListBinding.linearViewChapterAll.setOnClickListener { clickListener!!.onItemClickGenre(position) }
    }

    override fun getItemCount(): Int {
        return allGenreList.size
    }

    inner class ViewHolder(val itemListBinding: ItemListSelectChapterBinding) : RecyclerView.ViewHolder(itemListBinding.root)

    interface ClickGenreListener {
        fun onItemClickGenre(position: Int)
    }
}
